.changeset/big-schools-hang.md

```diff
@@ -0,0 +1,37 @@
+---
+'@xstate/store': major
+---
+
+Type parameters should now be explicitly provided to `createStore` and `createStoreWithProducer`. For sent and emitted events, the an `EventPayloadMap` should be provided, which is a map of event types to their payloads.
+
+```ts
+createStore<
+  // Context
+  {
+    count: number;
+  },
+  // Sent events
+  {
+    inc: {
+      by: number;
+    };
+  },
```

Comment:
This payload map doesn't feel like an ergonomic explicit type argument - you won't need it anywhere else than at the `createStore` call. A union plays better with other utils you might want to build around this API as a user. Like, even a `logEvent` becomes much more cumbersome to type because you need to somehow extract the event type from your store instead of just using the same union type that you could use here

Reply:
The payload map is mostly to still allow for inference when the type parameters are not provided.

Reply:
We no longer use explicit type arguments